OSDOCS-6864: Adjust pods per node to reflect OVN-Kubernetes limits (#62307)
openshift/openshift-docs: modules/openshift-cluster-maximums-major-releases.adoc

| Number of back-ends per service
| 5,000

| Number of deployments per namespace ^[6]^
| 2,000

| Number of build configs
| 12,000

| Number of custom resource definitions (CRD)
| 512 ^[8]^

|===
[.small]
--
1. Pause pods were deployed to stress the control plane components of {product-title} at 2000 node scale. The ability to scale to similar numbers will vary depending upon specific deployment and workload parameters.
2. The pod count displayed here is the number of test pods. The actual number of pods depends on the application's memory, CPU, and storage requirements.

> 3. This was tested on a cluster with 31 servers: 3 control planes, 2 infrastructure nodes, and 26 worker nodes. The default `maxPods` is still 250. To get to 2,500 pods per node, the cluster must be created with `maxPods` set to `2500` using a custom kubelet config. If you need 2,500 user pods, you need a `hostPrefix` of `20` because there are 10-15 system pods already running on the node. The maximum number of pods with attached persistent volume claims (PVC) depends on the storage backend from where PVC are allocated. In our tests, only {rh-storage} v4 (OCS v4) was able to satisfy 2,500 of pods per node.

Reviewer: Need some changes here.

"The maximum number of pods with attached persistent volume claims (PVC) depends on the storage backend from where PVC are allocated.  In our tests, only {rh-storage} v4 (OCS v4) was able to satisfy 2,500 of pods per node." Can be removed as it is irrelevant since I did not use OCS nor pods with PVCs in my tests.

I would swap the order of "The default maxPods..." and "... hostPrefix ..." sentences because hostPrefix is an install-time configuration and maxPods is a Day 2 post-install operation.

Are we planning to link directly to the blog from the docs?

https://cloud.redhat.com/blog/running-2500-pods-per-node-on-ocp-4.13

> 3. This was tested on a cluster with 32 servers: 1 jump host, 3 control planes, 2 infrastructure nodes, and 26 worker nodes. The default `maxPods` is still 250. To get to 2,500 pods per node, the cluster must be created with `maxPods` set to `2500` using a custom kubelet config. If you need 2,500 user pods, you need a `hostPrefix` of `21` because there are 10-15 system pods already running on the node. The maximum number of pods with attached persistent volume claims (PVC) depends on the storage backend from where PVC are allocated. In our tests, only {rh-storage} v4 (OCS v4) was able to satisfy 2,500 of pods per node.

Reviewer 2: 2500 pods would require a hostPrefix of /20

> 3. This was tested on a cluster with 32 servers: 1 jump host, 3 control planes, 2 infrastructure nodes, and 26 worker nodes. The default `maxPods` is still 250. To get to 2,500 pods per node, the cluster must be created with `maxPods` set to `2500` using a custom kubelet config. If you need 2,500 user pods, you need a `hostPrefix` of `21` because there are 10-15 system pods already running on the node. The maximum number of pods with attached persistent volume claims (PVC) depends on the storage backend from where PVC are allocated. In our tests, only {rh-storage} v4 (OCS v4) was able to satisfy 2,500 of pods per node.

Reviewer: `{rh-storage} v4 (OCS v4)` Unless there was someone else testing at this density, I did not have any PVCs or OCS in my setup.

Reviewer: I called this out in July. Why was did we keep it here?

> 3. This was tested on a cluster with 32 servers: 1 jump host, 3 control planes, 2 infrastructure nodes, and 26 worker nodes. The default `maxPods` is still 250. To get to 2,500 pods per node, the cluster must be created with `maxPods` set to `2500` using a custom kubelet config. If you need 2,500 user pods, you need a `hostPrefix` of `21` because there are 10-15 system pods already running on the node. The maximum number of pods with attached persistent volume claims (PVC) depends on the storage backend from where PVC are allocated. In our tests, only {rh-storage} v4 (OCS v4) was able to satisfy 2,500 of pods per node.

Reviewer 3: I wouldn't mention the jump host.

> 4. The maximum pods per node is 2,500 for clusters using the OVN-Kubernetes network plugin. The maximum pods per node for the OpenShift SDN network plugin is 500 pods.

Reviewer 4: The wording should reflect that it is the maximum tested pods per node. Ie changing it to:

The maximum tested pods per node is 2,500 for clusters using the OVN-Kubernetes network plugin. The maximum tested pods per node for the OpenShift SDN network plugin is 500 pods.

Reviewer 5: These are all tested cluster maximums anyway as per the title of the page.

> 4. The maximum pods per node is 2,500 for clusters using the OVN-Kubernetes network plugin. The maximum pods per node for the OpenShift SDN network plugin is 500 pods.

Reviewer 3: ```suggestion
4. The maximum pods per node is 2,500 for clusters using the OVNKubernetes network plugin. The maximum pods per node for the OpenShift SDN network plugin is 500 pods.
```

5. When there are a large number of active projects, etcd might suffer from poor performance if the keyspace grows excessively large and exceeds the space quota. Periodic maintenance of etcd, including defragmentation, is highly recommended to free etcd storage.

> 6. There are a number of control loops in the system that must iterate over all objects in a given namespace as a reaction to some changes in state. Having a large number of objects of a given type in a single namespace can make those loops expensive and slow down processing given state changes. The limit assumes that the system has enough CPU, memory, and disk to satisfy the application requirements.

Reviewer 6: ```suggestion
6. There are several control loops in the system that must iterate over all objects in a given namespace as a reaction to some changes in state. Having a large number of objects of a given type in a single namespace can make those loops expensive and slow down processing given state changes. The limit assumes that the system has enough CPU, memory, and disk to satisfy the application requirements.
```

> 7. Each service port and each service back-end has a corresponding entry in iptables. The number of back-ends of a given service impact the size of the endpoints objects, which impacts the size of data that is being sent all over the system.

Reviewer 6: ```suggestion
7. Each service port and each service back-end has a corresponding entry in `iptables`. The number of back-ends of a given service impact the size of the endpoint objects, which impacts the size of data that is sent over the system.
```
See markup guide: https://redhat-documentation.github.io/asciidoc-markup-conventions/

> 8. {product-title} has a limit of 512 total custom resource definitions (CRD), including those installed by {product-title}, products integrating with {product-title} and user created CRDs. If there are more than 512 CRDs created, then there is a possibility that `oc` commands requests may be throttled.

Reviewer 6: ```suggestion
8. {product-title} has a limit of 512 total custom resource definitions (CRD), including those installed by {product-title}, products integrating with {product-title} and user-created CRDs. If there are more than 512 CRDs created, then there is a possibility that `oc` command requests might be throttled.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m
